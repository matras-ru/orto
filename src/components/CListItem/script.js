import { selfInstall } from '@/utils';
import { mergeData } from 'vue-functional-data-merge';

import ThemeClass from '@/themes/default/CListItem';

const { baseClass } = ThemeClass;

const props = {
    tag: {
        type: String,
        default: 'li'
    }
};
export default {
    name: 'ListItem',

    functional: true,

    install(Vue, theme) {
        selfInstall(Vue, theme, this);
    },

    props,

    render(h, { props, data, children }) {
        const componentData = {
            staticClass: baseClass.join(' ')
        };

        return h(props.tag, mergeData(data, componentData), children);
    }
};
