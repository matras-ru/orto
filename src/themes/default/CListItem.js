export const base = '';
export const DISPLAY_INLINE = 'inline-block';
export const DISPLAY_BLOCK = 'block';

export default {
    base,
    DISPLAY_INLINE,
    DISPLAY_BLOCK
};
