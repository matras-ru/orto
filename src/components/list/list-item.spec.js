/* eslint-env jest */
import { mount } from '@vue/test-utils';
import CListItem from './list-item';

const defaultClass = 'inline-block';

describe('CListItem', () => {
    it('default empty CListItem is functional component and rendered', () => {
        const wrapper = mount(CListItem);
        expect(wrapper.isFunctionalComponent).toBe(true);

        expect(wrapper.element.tagName).toEqual('LI');
        expect(wrapper.classes().sort()).toEqual(`${defaultClass}`.split(' ').sort());
        expect(wrapper.classes().length).toBe(1);
        expect(wrapper.text()).toEqual('');

        expect(wrapper.html()).toMatchSnapshot();

        wrapper.destroy();
    });

    it('html tag equal div when prop tag = div', () => {
        const wrapper = mount(CListItem, {
            context: {
                props: {
                    tag: 'div'
                }
            }
        });

        expect(wrapper.element.tagName).toEqual('DIV');
        expect(wrapper.html()).toMatchSnapshot();

        wrapper.destroy();
    });

    it('custom attributes', () => {
        const wrapper = mount(CListItem, {
            attrs: {
                role: 'presentation'
            }
        });

        expect(wrapper.attributes('role')).toBe('presentation');
    });
});
