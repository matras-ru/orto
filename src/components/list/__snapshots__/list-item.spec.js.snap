// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CListItem default empty CListItem is functional component and rendered 1`] = `<li class="inline-block"></li>`;

exports[`CListItem html tag equal div when prop tag = div 1`] = `<div class="inline-block"></div>`;
